File: backend/app/schemas/health.py
import uuid
from datetime import datetime
from pydantic import BaseModel, ConfigDict

class HealthSnapshotResponse(BaseModel):
    overall_score: int
    category_scores: dict[str, int]
    reasons: list[str]
    computed_at: datetime
    previous_score: int | None = None

    model_config = ConfigDict(from_attributes=True)


class HealthSnapshotListResponse(BaseModel):
    items: list[HealthSnapshotResponse]
    total: int
    limit: int
    offset: int

File: frontend/src/lib/types/health.ts
export type HealthCategory =
    | "ci_cd"
    | "dependencies"
    | "security"
    | "issues"
    | "pull_requests"
    | "code_quality";

export interface RepositoryHealthSnapshot {
    overall_score: number;
    category_scores: Record<HealthCategory, number>;
    reasons: string[];
    computed_at: string;
    previous_score: number | null;
}

// Re-using RepositoryHealthSnapshot for history items as it matches exactly
export type HealthSnapshotHistoryItem = RepositoryHealthSnapshot;

export interface HealthSnapshotListResponse {
    items: HealthSnapshotHistoryItem[];
    total: number;
    limit: number;
    offset: number;
}
